### PR TITLE
Fix wallet sync starting on app launch

### LINKED ISSUE #107: When the app is launched syncing doesn't start for wallets until you navigate to the wallets tab
Image

### PR DESCRIPTION
**Pull request by CPunch**
closes #107 

kind of wanted to give a shot at this, all of the wallet sync code is in wallet_selector_page.go, would it make sense to split it out or would a hacky fix like this be suitable?

### CODE REVIEW

**Comment by crux25** [comment]
It is better to split the code, move the sync startup logic to home page. Calling wallet selector's navigation function while on home page will potentially introduce bugs in future, devs will assume that `OnNavigatedTo()` and  `OnNavigatedFrom()` are called only when the page is created and destroyed.

File: ui/page/root/home_page.go [hunk 1]
```diff
@@ -102,6 +102,11 @@ func (hp *HomePage) OnNavigatedTo() {
 	hp.ctx, hp.ctxCancel = context.WithCancel(context.TODO())
 
 	if hp.CurrentPage() == nil {
+		// start wallet sync
+		pg := NewWalletSelectorPage(hp.Load)
+		pg.OnNavigatedTo()
+		pg.OnNavigatedFrom()
```
Comment (Sirmorrison) [inline ui/page/root/home_page.go:108]: This is really hacky lol.

The only thought coming to mind is to load the wallets, and have them sync in background pending when we view the loaded wallets on the wallets page.

**Comment by dreacot** [comment]
@CPunch please fix branch conflicts

File: ui/page/root/home_page.go [hunk 2]
```diff
@@ -407,3 +450,104 @@ func (hp *HomePage) notificationSettingsLayout(gtx C) D {
...
+				// reset ticker
+				ticker.Reset(duration)
+			}
+			// Increase the counter
+			count++
+			log.Debugf("Attempting to check for internet connection in %s", duration.String())
+
+			go func() {
+				hp.isConnected = libutils.IsOnline()
+			}()
```
Comment (ukane-philemon) [inline ui/page/root/home_page.go:521]: I don't think there’s a need for the nested goroutine.

File: ui/page/root/home_page.go [hunk 3]
```diff
@@ -85,6 +92,21 @@ func NewHomePage(l *load.Load) *HomePage {
...
+	}()
+
+	// init shared page functions
+	toggleSync := func(unlock load.NeedUnlockRestore) {
+		if hp.WL.SelectedWallet.Wallet.IsConnectedToNetwork() {
+			go hp.WL.SelectedWallet.Wallet.CancelSync()
+			unlock(false)
+		} else {
+			hp.startSyncing(hp.WL.SelectedWallet.Wallet, unlock)
+		}
```
Comment (ukane-philemon) [inline ui/page/root/home_page.go:106]: I think this check is inverted.

**Review by ukane-philemon — COMMENTED** [review]
Quite straight forward.

**Comment by ukane-philemon** [comment]
Looks good, you can update the title.

**Review by ukane-philemon — CHANGES_REQUESTED** [review]
Almost.

**Comment by dreacot** [comment]
@CPunch you have branch conflicts

File: ui/page/root/home_page.go [hunk 4]
```diff
@@ -151,7 +151,6 @@ func (hp *HomePage) OnNavigatedTo() {
 		}
 	}
 
-	hp.CurrentPage().OnNavigatedTo()
```
Comment (CPunch) [inline ui/page/root/home_page.go]: looks like this caused a crash else where. when this line is included, we get a panic at 

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x134e042]

goroutine 45 [running]:
github.com/crypto-power/cryptopower/ui/page/root.(*OverviewPage).listenForMixerNotifications.func1()
        /home/cpunch/projects/cryptopower/ui/page/root/overview_page.go:878 +0x2c2
created by github.com/crypto-power/cryptopower/ui/page/root.(*OverviewPage).listenForMixerNotifications in goroutine 37
        /home/cpunch/projects/cryptopower/ui/page/root/overview_page.go:848 +0x6f8
```

File: ui/page/root/home_page.go [hunk 5]
```diff
@@ -405,6 +450,109 @@ func (hp *HomePage) notificationSettingsLayout(gtx C) D {
...
+	}
+	unlock(true)
+
+	if hp.isConnected {
+		// once network connection has been established proceed to
+		// start the wallet sync.
+		if err := wallet.SpvSync(); err != nil {
+			log.Debugf("Error starting sync: %v", err)
+		}
+		return
```
Comment (ukane-philemon) [inline ui/page/root/home_page.go:466]: Missed this. I think we should fall through (i.e remove the `return`) so we get to try again after the first 10 sec until we successfully start syncing.

**Review by ukane-philemon — CHANGES_REQUESTED** [review]
@CPunch, I think this PR is GTG after this.

File: ui/page/root/home_page.go [hunk 6]
```diff
@@ -405,6 +449,109 @@ func (hp *HomePage) notificationSettingsLayout(gtx C) D {
 	)
 }
 
+func (hp *HomePage) startSyncing(wallet sharedW.Asset, unlock load.NeedUnlockRestore) {
+	// Watchonly wallets do not have any password neithers need one.
```
Comment (ukane-philemon) [inline ui/page/root/home_page.go:453]: nit: neithers -> neither

**Review by ukane-philemon — APPROVED** [review]

**Review by dreacot — APPROVED** [review]

**Pull request merged**